arangod/Agency/MoveShard.cpp

```diff
@@ -939,13 +1016,19 @@ arangodb::Result MoveShard::abort(std::string const& reason) {
                          {
                            VPackArrayBuilder guard(&trx);
                            trx.add(VPackValue(_from));
-                           for (VPackSlice srv : VPackArrayIterator(plan)) {
-                             // from could be in plan as <from> or <_from>. Exclude to server always.
-                             if (srv.isEqualString(_from) || srv.isEqualString("_" + _from) ||
-                                 srv.isEqualString(_to)) {
-                               continue;
+                           if (plan.isArray()) {
+                             for (VPackSlice srv : VPackArrayIterator(plan)) {
+                               // from could be in plan as <from> or <_from>. Exclude to server always.
+                               if (srv.isEqualString(_from) || srv.isEqualString("_" + _from) ||
+                                   srv.isEqualString(_to)) {
+                                 continue;
+                               }
+                               trx.add(srv);
                              }
-                             trx.add(srv);
+                           } else {
+                             LOG_TOPIC("2e7b9", WARN, Logger::SUPERVISION) 
+                               << "failed to iterate over planned servers for shard "
+                               << _shard << " or a clone";
```

Comment:
It is quite serious if we find a non-array here, since this means that the world of `distributeShardsLike` has been violated. I think we should at least put a `TRI_ASSERT(false);` here and probably also not proceed with the transaction in the non-maintainer case as before.

Reply:
yes

Reply:
You did not address my second point: I do not think we should proceed with the transaction in the non-maintainer case, if some plan was not an array. Please at a `failed` and do not proceed if `failed`is `true`.

Reply:
done